src/intrinsic/annotations.rs

```diff
@@ -65,19 +71,57 @@ impl AnnotationError {
                         "There must not be any statements after this annotated statement (and the annotated statement must not be nested in a block).",
                     ))
             }
+            AnnotationError::CalculusEncodingMismatch(span, calculus_name, encoding_name ) => {
+                Diagnostic::new(ReportKind::Error, span)
+                    .with_message(format!(
+                        "The '{}' calculus does not support the '{}' encoding.",
```

Comment:
Maybe also include the direction? E.g. using `@invariant` with `@wlp` in a `coproc` causes confusion because the diagnostic "The 'wlp' calculus does not support the 'invariant' encoding." makes it look like the invariant-rule cannot be used for wlp at all.